x-pack/plugins/actions/server/builtin_action_types/lib/request_oauth_jwt_token.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import qs from 'query-string';
+import axios from 'axios';
+import stringify from 'json-stable-stringify';
+import { request } from '../lib/axios_utils';
+import { Logger } from '../../../../../../src/core/server';
+import { ActionsConfigurationUtilities } from '../../actions_config';
+
+export const OAUTH_JWT_BEARER_GRANT_TYPE = 'urn:ietf:params:oauth:grant-type:jwt-bearer';
+
+export interface OAuthJWTTokenResponse {
+  scope?: string;
+  tokenType: string;
+  expiresIn: number;
+  accessToken: string;
+}
+
+export async function requestOAuthJWTToken(
```

Comment:
Seems like this is very similar to `requestOAuthClientCredentialsToken` with the addition of an `assertion` and a different grant type. Would it make sense to update that function to be more generic?

Reply:
Yeah, this is a great idea. Thank you @ymao1 !